src/sssom/writers.py

```diff
@@ -99,20 +99,20 @@ def write_rdf(
     print(t.decode(), file=file)
 
 
-# todo: not sure the need for serialization param here; seems superfluous for some of these funcs
-def write_fhir_json(msdf: MappingSetDataFrame, output: TextIO, serialisation="fhir") -> None:
-    """Write a mapping set dataframe to the file as FHIR ConceptMap JSON."""
-    data = to_fhir_json(msdf)
-    json.dump(data, output, indent=2)
-
-
 def write_json(msdf: MappingSetDataFrame, output: TextIO, serialisation="json") -> None:
```

Comment:
## Condensed 3 `write_json()` functions into 1
If you look at the diffs, it is observed that these 3 functions for formats `json`, `fhir_json`, and `ontoportal_json` had the same pattern. For example, each had had an either unused or superfluous param `serialisation`. I've condensed these into 1 function, which seems to be the original intent, and also DRYs things up.

I'm happy to revert this if it wasn't the original intent / this is not liked.

### Potential other improvements
`func_map` could probably be turned into a constant, like `WRITER_FUNCTIONS`. Which also seems redundant though, and makes me question the change I just made here.

Reply:
Very nice, I definitely like it. 

However, you need to preserve the old methods, annotate them as "deprecated" and wrap the new one., like

```
@deprecated
def write_fhir_json(msdf: MappingSetDataFrame, output: TextIO, serialisation="fhir") -> None:
    """Write a mapping set dataframe to the file as FHIR ConceptMap JSON."""
    write_json(msdf, output, "fhir_json")
```

This is pseudo code - the goal is to not break code that needs those methods.

Reply:
- [x] Done!

<details><summary>Details</summary>
<p>

```py
@deprecated(deprecated_in="0.4.7", details="Use write_json() instead")
def write_fhir_json(msdf: MappingSetDataFrame, output: TextIO, serialisation="fhir_json") -> None:
    """Write a mapping set dataframe to the file as FHIR ConceptMap JSON."""
    if serialisation != "fhir_json":
        raise ValueError(
            f"Unknown json format: {serialisation}, currently only fhir_json supported"
        )
    write_json(msdf, output, serialisation="fhir_json")


@deprecated(deprecated_in="0.4.7", details="Use write_json() instead")
def write_ontoportal_json(
    msdf: MappingSetDataFrame, output: TextIO, serialisation: str = "ontoportal_json"
) -> None:
    """Write a mapping set dataframe to the file as the ontoportal mapping JSON model."""
    if serialisation != "ontoportal_json":
        raise ValueError(
            f"Unknown json format: {serialisation}, currently only ontoportal_json supported"
        )
    write_json(msdf, output, serialisation="ontoportal_json")
```

</p>
</details>